src/system.rs

```diff
@@ -0,0 +1,173 @@
+use crate::errors::ETrackedPropertyError;
+use crate::{sys, Context, TrackedDeviceIndex};
+
+use std::ffi::{CString, CStr};
+use std::marker::PhantomData;
+use std::pin::Pin;
+use std::ptr::null_mut;
+
+pub struct SystemManager<'c> {
+    ctx: PhantomData<&'c Context>,
+    inner: Pin<&'c mut sys::IVRSystem>,
+    string_buf: Vec<u8>,
+}
+
+mod private {
+    pub trait Sealed {}
+}
+
+
+type PropResult<T> = Result<T, ETrackedPropertyError>;
+
+/// Trait implemented by types that represent storage types of properties.
+pub trait TrackedDevicePropertyValue: private::Sealed + Sized {
```

Comment:
```suggestion
pub trait TrackedDeviceProperty: private::Sealed + Sized {
```
The name of this trait is still confusing. You are returning this when you call `get_tracked_device_property`. So it *is* a `TrackedDeviceProperty`. If you have a different intent you wish to add later after this MVP, such as a way to differentiate between the rust-native types and the types that are actually stored in openvr, we can add that in a later PR. For now, I will apply some suggestions to this and merge.

Reply:
I'm thinking about this as a key value map. The property is the key, the value of that property is the value. I guess you're thinking of it as the name of the property being the key, and the property itself being a value?

Reply:
To expand: I do not consider a Matrix3x4 or a u64 to be a property. A CurrentTrackingUniverse is a property.

Reply:
yes, precisely. I know you may revise this in a later PR though, and I'm 100% open to that (with appropriate discussion, of course)

Reply:
If you want to separate it into PropertyNames being the key and Properties being the value, fine, but we need nanes for both concepts, and I'd prefer them to make sense.

Reply:
Lets please punt this to a later PR.

Reply:
Fine. Go ahead and merge it then.